cmd/containerd-shim-runc-v2/process/exec.go

```diff
@@ -204,7 +204,11 @@ func (e *execProcess) start(ctx context.Context) (err error) {
 	if socket != nil {
 		opts.ConsoleSocket = socket
 	}
-	if err := e.parent.runtime.Exec(ctx, e.parent.id, e.spec, opts); err != nil {
+	// generate a special log file for every exec process.
+	e.parent.runtime.Log = filepath.Join(filepath.Dir(e.parent.runtime.Log), fmt.Sprintf("%s-exec.log", e.id))
+	err = e.parent.runtime.Exec(ctx, e.parent.id, e.spec, opts)
+	defer os.RemoveAll(e.parent.runtime.Log)
```

Comment:
If we remove the log after this method returns, then does that mean we're just writing and removing files that will never be seen? Is this essentially the same thing as setting `e.parent.runtime.Log = /dev/null`? Where else could a user see their exec's logs?

Reply:
shim will call getLastRuntimeError to read err log  in  e.parent.runtimeError(err, "OCI runtime exec failed")  and send this log to containerd, user can see this log in containerd log.